experimental/ModelBuilder/test/BenchMatMulVectorGPU.cpp

```diff
@@ -0,0 +1,233 @@
+                .setInterchange({1, 0, 2}))
+        .setHoistInvariantCode(false)
+        .vectorize<linalg::MatmulOp>()
+        .unrollVector<vector::ContractionOp>(
+            {cooperativeMatrixM, cooperativeMatrixN, cooperativeMatrixK});
+    modelBuilder.getModuleRef()->walk(
+        [&](FuncOp fn) { strategy.transform(fn); });
+    addLoweringPasses(pm, {resRows / tileM, resColumns / tileN, 1},
+                      {typeA, typeB, typeC});
+  };
+  runner.compile(options, {vulkanWrapper});
+
+  // 3. Allocate data within data structures that interoperate with the MLIR ABI
+  // conventions used by codegen.
+  auto oneInit = [](unsigned idx, uint8_t *ptr) { ptr[idx] = 2 * idx + 1; };
+  auto incInit = [](unsigned idx, uint8_t *ptr) { ptr[idx] = idx; };
+  auto zeroInit = [](unsigned idx, uint32_t *ptr) { ptr[idx] = 0; };
+  auto A = makeInitializedStridedMemRefDescriptor<uint8_t, 2>(
+      {resRows, reductionSize}, oneInit);
+  auto B = makeInitializedStridedMemRefDescriptor<uint8_t, 2>(
+      {reductionSize, resColumns}, incInit);
+  auto C = makeInitializedStridedMemRefDescriptor<uint32_t, 2>(
+      {resRows, resColumns}, zeroInit);
+  auto CPURes = makeInitializedStridedMemRefDescriptor<uint32_t, 2>(
+      {resRows, resColumns}, zeroInit);
+
+  // Is checking corretness compare to the value computed on CPU.
+  if (correctness) {
+    for (int i = 0; i < resRows; i++) {
+      for (int j = 0; j < resColumns; j++) {
+        uint32_t acc = (*C)[i][j];
+        for (int k = 0; k < reductionSize; k++) {
+          uint32_t a = (*A)[i][k];
+          uint32_t b = (*B)[k][j];
+          acc += a * b;
+        }
+        (*CPURes)[i][j] = acc;
+      }
+    }
+  }
+
+  // 4. Call the funcOp named `funcName`.
+  auto err = runner.invoke(std::string(funcName) + "_wrapper", A, B, C);
+  if (err) llvm_unreachable("Error running function.");
+
+  if (correctness) {
+    bool correct = true;
+    for (int i = 0; i < resRows; i++) {
+      for (int j = 0; j < resColumns; j++) {
+        if ((*CPURes)[i][j] != (*C)[i][j]) {
+          correct = false;
+          printf("mismatch at index(%i, %i) was expecting %i but got %i\n", i,
+                 j, (*CPURes)[i][j], (*C)[i][j]);
+        }
+      }
+    }
+    if (correct) printf("pass\n");
+  }
+}
+
+int main(int argc, char **argv) {
+  ModelBuilder::registerAllDialects();
+  iree::Initializer::RunInitializers();
+  // Allow LLVM setup through command line and parse the
+  // test specific option for a runtime support library.
+  llvm::InitLLVM y(argc, argv);
+  llvm::cl::ParseCommandLineOptions(argc, argv, "BenchMatMulVectorGPU\n");
+  int tileK = 32;
+  for (int tileM = 8; tileM <= 128; tileM *= 2) {
+    for (int tileN = 8; tileN <= 128; tileN *= 2) {
+      int m = 4096;
+      int n = 4096;
+      int k = 4096;
+      if (correctness) {
+        m = 256;
+        n = 256;
+        k = 256;
+      }
+      printf("tileM=%i tileN=%i tileK=%i\n", tileM, tileN, tileK);
```

Comment:
Might worth to also print out m, n, k once at the very beginning.

Reply:
Done.